geoindicators/src/main/groovy/org/orbisgis/RsuIndicators.groovy

```diff
@@ -619,4 +615,164 @@ static IProcess rsuEffectiveTerrainRoughnessHeight() {
+                            caseQueryDistrib += "SUM(CASEWHEN(azimuth>=${d-angleRangeSize} AND azimuth<$d, length, 0)) AS " +
+                                    "rsu_road_direction_distribution_d${d-angleRangeSize}_$d,"
+                        }
+                    }
+                    // If only certain levels are considered independantly
+                    else{
+                        ifZindex = ", zindex "
+                        levelConsiderated.each { lev ->
+                            caseQueryDens += "SUM(CASEWHEN(zindex = $lev, ST_LENGTH(the_geom), 0))/rsu_area " +
+                                    "AS rsu_linear_road_density_h${lev.toString().replaceAll('-','minus')},"
+                            nameDens.add("rsu_linear_road_density_h${lev.toString().replaceAll('-','minus')}")
+                            for (int d=angleRangeSize; d<=180; d+=angleRangeSize){
+                                caseQueryDistrib += "SUM(CASEWHEN(azimuth>=${d-angleRangeSize} AND azimuth<$d AND " +
+                                        "zindex = $lev, length, 0)) AS " +
+                                        "rsu_road_direction_distribution_h${lev.toString().replaceAll("-","minus")}" +
+                                        "_d${d-angleRangeSize}_$d,"
+                            }
+                        }
+                    }
+
+                    // QUERIES EXECUTED DEPENDING ON THE CONTENT OF THE OPERATION LIST
+                    // If the road direction distribution is calculated, explode the roads into segments in order to calculate
+                    // their length for each azimuth range
+                    if (operations.contains("rsu_road_direction_distribution")) {
+                        def queryExpl = "DROP TABLE IF EXISTS $roadExpl;" +
+                                "CREATE TABLE $roadExpl AS SELECT id_rsu, the_geom, " +
+                                "CASEWHEN(ST_AZIMUTH(ST_STARTPOINT(the_geom), ST_ENDPOINT(the_geom))>=pi()," +
+                                "ROUND(DEGREES(ST_AZIMUTH(ST_STARTPOINT(the_geom), ST_ENDPOINT(the_geom))))-180," +
+                                "ROUND(DEGREES(ST_AZIMUTH(ST_STARTPOINT(the_geom), ST_ENDPOINT(the_geom))))) AS azimuth,"+
+                                "ST_LENGTH(the_geom) AS length $ifZindex FROM ST_EXPLODE('(SELECT ST_TOMULTISEGMENTS(the_geom)" +
+                                " AS the_geom, id_rsu $ifZindex FROM $roadInter)');"
+                        // Calculate the road direction for each direction and optionally level
+                        def queryDistrib = queryExpl+"CREATE TABLE $roadDistrib AS SELECT id_rsu, "+caseQueryDistrib[0..-2]+
+                                " FROM $roadExpl GROUP BY id_rsu;"+
+                                "CREATE INDEX IF NOT EXISTS id_u ON $rsuTable($idColumnRsu);"+
+                                "CREATE INDEX IF NOT EXISTS id_d ON $roadDistrib(id_rsu);"+
```

Comment:
$roadDistrib($idColumnRsu) ?

Reply:
In order to simplify the code reading and writing, I use specific names (that are the names corresponding to the Abstract tables) for the temporary tables.
I actually do not know what is the best solution: keeping the variables all along the code, using only the Generic table field names all along the code or doing a mix of both (corresponding to what I am currently doing).